src/components/mdx/code-block.js

```diff
@@ -19,9 +19,10 @@ const codeBlockClassName = theme => {
 }
 
 const addNewlineToEmptyLine = line => {
-  if (line && line.length === 1 && line[0].empty) {
-    // Improves copy/paste behavior
-    line[0].content = "\n"
+  // addresses pasting bug on firefox by creating a new line at the end of a code block row
+  
+  if ( line && (line[line.length-1].content.trim() === "" || line[0].content.trim() === "")) {
```

Comment:
We should make it automatic on pre-commit, but you should run your code through a formatter. Lots of existing conventions here have been broken.